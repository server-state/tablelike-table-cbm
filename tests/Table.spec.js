import React from "react";
import { create } from "react-test-renderer";
import Table from "../component/src";

describe("Table CBM", () => {
    test("Invalid data", () => {
        console.error = jest.fn();
        const cbm = create(<Table />);
        expect(cbm).toMatchSnapshot();
        expect(console.error).toBeCalled();
    });

    test('Valid data', () => {
        const mockData = {
            _fields: ['fielda', 'fieldb'],
            rows: [
                {
                    fielda: 'Hello world',
                    fieldb: 1
                },
                {
                    fielda: 'Test',
                    fieldb: 2
                }
            ]
        };
        const cbm = create(<Table data={mockData} onRefresh={() => {}} onRegisterAction={() => {}}/>)
        expect(cbm).toMatchSnapshot();
    })
});
