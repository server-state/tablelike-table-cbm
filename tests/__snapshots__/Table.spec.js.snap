// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table CBM Invalid data 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  Invalid data type. Expected TableLike data to get passed to the visualization.
</p>
`;

exports[`Table CBM Valid data 1`] = `
<table
  className="MuiTable-root"
>
  <thead
    className="MuiTableHead-root"
  >
    <tr
      className="MuiTableRow-root MuiTableRow-head"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        fielda
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        fieldb
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root"
  >
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Hello world
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        1
      </td>
    </tr>
    <tr
      className="MuiTableRow-root"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        Test
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body"
      >
        2
      </td>
    </tr>
  </tbody>
</table>
`;
